subsys/shell/shell.c

```diff
@@ -1366,19 +1375,15 @@ int shell_prompt_change(const struct shell *shell, const char *prompt)
 
 void shell_help(const struct shell *shell)
 {
-	__ASSERT_NO_MSG(shell);
-
-	if (!IS_ENABLED(CONFIG_SHELL_HELP)) {
-		return;
-	}
-
-	shell_help_cmd_print(shell);
-	shell_help_subcmd_print(shell);
+	k_mutex_lock(&shell->ctx->wr_mtx, K_FOREVER);
+	shell_internal_help_print(shell);
+	k_mutex_unlock(&shell->ctx->wr_mtx);
 }
 
 int shell_execute_cmd(const struct shell *shell, const char *cmd)
 {
 	u16_t cmd_len = shell_strlen(cmd);
```

Comment:
please add assert to check if not in shell context

Reply:
done